scene/gui/control.cpp

```diff
@@ -3929,8 +3929,11 @@ void Control::_notification(int p_notification) {
 
 		case NOTIFICATION_DRAW: {
 			_update_canvas_item_transform();
-			RenderingServer::get_singleton()->canvas_item_set_custom_rect(get_canvas_item(), !data.disable_visibility_clip, Rect2(Point2(), get_size()));
-			RenderingServer::get_singleton()->canvas_item_set_clip(get_canvas_item(), data.clip_contents);
+
+			RID ci = get_canvas_item();
+			RenderingServer::get_singleton()->canvas_item_set_custom_rect(ci, !data.disable_visibility_clip, Rect2(Point2(), get_size()));
+			RenderingServer::get_singleton()->canvas_item_set_merge_custom_rect(ci, !data.disable_visibility_clip);
```

Comment:
I'm not sure about the knock-on effects of doing this. I think any control nodes rely on having precise control over their rect on the scene-side. This would introduce a situation where the scene is no longer aware of the size of the rect. 

I am thinking it may cause problems with things like containers that need to auto-balance their contents, or anchors which need to dynamically adjust to the size of the control node.

My gut feeling is that this change will have very far reaching consequences despite seemingly just being a fix for the `clip_children` setting. 

Maybe this should work more like what we do with the CanvasGroup node? Basically you can detect if a Canvasitem is a CanvasGroup owner, and then only change the rect size if it is